### ISSUE TITLE
Issuing RPC commands to a specified rpc socket file without knowing its network

### ISSUE THREAD

**Issue by shesek** [ElementsProject/lightning#3352]
I'm trying to issue RPC commands using `lightning-cli` against an rpc socket file at a known location, specified with `--rpc-file`. It seems like even then, c-lightning still expects `lightning-dir` and `network` to be provided, and fails if they're invalid (when trying to move into them).

Is there a reason they're necessary? Are they used for anything other than determining the path of the socket file?

**Comment by rustyrussell** [comment]
Yes, it now needs to know the network (like bitcoin-cli). But yes, we should allow you to specify rpc-file directly and not care.

**Comment by shesek** [comment]
For anyone looking for a workaround to make this work with v0.8.0, you can just give c-lightning a dummy base directory with an empty network subdirectory to avoid the `chdir` error:

```bash
$ mkdir /tmp/dummy /tmp/dummy/bitcoin
$ lightning-cli --lightning-dir /tmp/dummy --rpc-file /path/to/lightning-rpc getinfo
```

(I ended up [doing this for spark](https://github.com/shesek/spark-wallet/blob/f1a684f9e52e470229cb421518f733e602246035/scripts/docker-entrypoint.sh#L85-L88))

**Comment by shesek** [comment]
Fixed by #3353, closing.

**Comment by ZmnSCPxj** [comment]
We normally close only once the pull request is actually merged into master.  This is automatically done by using `fixes` keyword in the pull request description or commit. Reopening.
